### PR TITLE
Add support for WebSockets API Terraform packaging

### PR DESCRIPTION
**Pull request by adityamohta**
*Issue https://github.com/aws/chalice/issues/1670

*Description of changes:*

Adds terraform packaging support for WebSockets API with its unit tests.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by adityamohta** [comment]
We love chalice and actively use it with Terraform deployment in some of our services. 

We found support for websockets_api deployment with `chalice deploy` and `SAM`. But, `terraform` support (which we really love) was missing. 

Therefore, we decided to add support for the same. 

---

Since this is my first contribution to Chalice, it would be great if a maintainer can take out time to run a workflow/review this PR. 

Perhaps @jamesls ?

File: chalice/package.py [hunk 1]
```diff
@@ -858,17 +857,198 @@ def _generate_managediamrole(self, resource, template):
...
+        if 'websocket_connect' in aws_lambda_functions:
+            output['WebsocketConnectHandlerArn'] = {
+                "value": "${aws_lambda_function.websocket_connect.arn}"}
+            output['WebsocketConnectHandlerName'] = {
+                "value": "${aws_lambda_function.websocket_connect}"}
+        if 'websocket_message' in aws_lambda_functions:
+            output['WebsocketMessageHandlerArn'] = {
+                "value": "${aws_lambda_function.websocket_message.arn}"}
+            output['WebsocketMessageHandlerName'] = {
+                "value": "${aws_lambda_function.websocket_message}"}
```
Comment (jamesls) [inline chalice/package.py:978]: I noticed this prints out all the properties of the lambda function: 

```
WebsocketMessageHandlerName = {
  "architectures" = tolist([
    "x86_64",
  ])
  "arn" = "arn:aws:lambda:us-west-2:12345:function:testws-dev-websocket_message"
  "code_signing_config_arn" = ""
  "dead_letter_config" = tolist([])
  "description" = ""
  "environment" = tolist([])
```

The SAM/cfn version uses `{'Ref': 'WebsocketMessage'}` which just returns the [resource name](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-lambda-function.html#aws-resource-lambda-function-return-values).

I believe we just need to add a `.function_name` to this output (same to all the other ws function handlers).

File: chalice/package.py [hunk 2]
```diff
@@ -101,8 +101,8 @@ def service_principal(self, service):
 class ResourceBuilder(object):
     def __init__(self,
                  application_builder,  # type: ApplicationGraphBuilder
-                 deps_builder,         # type: DependencyBuilder
-                 build_stage,          # type: BuildStage
+                 deps_builder,  # type: DependencyBuilder
```
Comment (jamesls) [inline chalice/package.py:104]: From what I can tell, `prcheck` still passes without these formatting changes.  I'm going to try reverting these changes and see if the CI passes.

**Review by jamesls — COMMENTED** [review]
Changes look good.  I had some small feedback, should be pretty straightforward so I'll go ahead and incorporate it real quick.

**Comment by jamesls** [comment]
I wasn't able to push to the branch for this PR for some reason so I incorporated the feedback and opened up a new PR: https://github.com/aws/chalice/pull/1928.  Let's track the remaining work over in that PR.

File: chalice/package.py (hunk 1 above)
Comment (adityamohta) [reply chalice/package.py:978]: Nice catch! Thanks for fixing it @jamesls

**Pull request closed without merging**